### ISSUE TITLE
Development Environment creation causing breakages

### ISSUE THREAD

**Issue by mattAux** [Shopify/shopify-app-cli#317]
# Issue summary

Jumped back into a project after 2 weeks to set up some deev / prod environments & seems to struggle to load `index.js`
I'm experiencing extremely long load times for my index page & often it errors out.
I didn't experience any of these issues prior to attaching this codebase to another app for development purposes.

Sorry if this is a novice / trivial issue.

## Actual behavior

Running shopify serve then going to the index page is returning the following in my terminal.
```
[ event ] build page: /
[ wait ]  compiling ...
[ event ] disposing inactive page(s): / 
[ event ] client pings, but there's no entry for page: /
[ event ] build page: /
[ wait ]  compiling ...
   Error: write EPIPE
       at WriteWrap.afterWrite (net.js:868:14)
   Error: write EPIPE
       at WriteWrap.afterWrite (net.js:868:14)
   Error: write EPIPE
      at WriteWrap.afterWrite (net.js:868:14)
   Error: write EPIPE
       at WriteWrap.afterWrite (net.js:868:14)
```

In the console on the index page of my app i'm seeing the following.
![image](https://user-images.githubusercontent.com/44592071/65649775-1676ac80-e05c-11e9-8edc-a8142919e759.png)

When I save a generated page - the app doesn't compile anything & I can't see any changes.
My terminal return the below
```
┃ [ event ] build page: /UploadCsv
┃ [ wait ]  compiling ...
┃
┃   Error: write EPIPE
┃       at WriteWrap.afterWrite (net.js:868:14)
┃
┃ [ event ] disposing inactive page(s): /UploadCsv
┃ [ event ] build page: /UploadCsv
┃ [ wait ]  compiling ...
┃ [ wait ]  compiling ...
hello
┃ [ wait ]  compiling ...
┃
┃   Error: read ECONNRESET
┃       at TCP.onread (net.js:622:25)
┃
┃
┃   Error: write EPIPE
┃       at WriteWrap.afterWrite (net.js:868:14)
┃
┃
┃   Error: write EPIPE
┃       at WriteWrap.afterWrite (net.js:868:14)
┃

```
And after some time I see this in my console.
![image](https://user-images.githubusercontent.com/44592071/65652278-52fad600-e065-11e9-80df-65ca56734d64.png)

Spinning up a new app using `shopify generate project` is still giving me a long load time :(
However, my changes are compiled correctly & I can actually work on it..

I'm worried that I may have caused these breakages when trying to create development / live environments.

My approach was to have two apps. 

**SomeApp**: Prod
This app has the production Urls set up in the Shopify Partners app setup panel ( heroku etc )

**DevSomeApp**: Development
This app would have only Ngrok Urls & would use this to build / test improvements.

**Both apps live off the same codebase & will deploy master when changes are ready to be pushed live**

Is there some other way to handle this that I haven't been able to pick up on?
Has changing out the .env variables to suit my dev-app caused these issues?
How do I set up a development environment without affecting the live?

## Specifications

- App type: Node
- Operating System: macOS Mojave v: 10.14.6 (18G95)

**Comment by katiedavis** [comment]
Hi @mattAux I'm not sure I'm following. Can you clarify 'attaching this codebase to another app for development purposes'?
Those errors are runtime errors, but I'm not sure they're related the the Shopify-app-cli.

**Comment by katiedavis** [comment]
Hey @mattAux I haven't heard back from you so I'm going to close this one for now :)

**Comment by mattAux** [comment]
Hey @katiedavis my apologies for the late response! Have been away.
So the issue was with the CLI. 
Running Shopify serve resulted in a huge wait time on initial compile & any saves resulted in the same.
I presumed it may be my project causing the issues - so I spun up a new project & ran into the same long load time issues. 
I also got a colleague to try the same.

I ended up recreating the app outside of the CLI - simply because waiting 2+ minutes for initial compile & failing compiles after a save wasn't going to be a good development environment!

I'll have another crack at spinning up an app & see how it goes.
